### ISSUE TITLE
Accessibility subsite in Design documentation

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/documentation#788]
# Accessibility

We track all issues and enhancements regarding accessibility with the [**feature: accessibility** label]( https://github.com/nextcloud/server/labels/feature%3A%20accessibility). They are also all part of the **design** label.

The posters on [dos and don'ts on designing for accessibility](https://accessibility.blog.gov.uk/2016/09/02/dos-and-donts-on-designing-for-accessibility/) are a great summary. Print out [the PDF](https://github.com/UKHomeOffice/posters/blob/master/accessibility/dos-donts/posters_en-UK/accessibility-posters-set.pdf) and hang them up. :)

## Screenreader accessibility

- Images should always have an `alt` attribute describing their function, like `alt="App menu"` (don’t forget the translations). If the image is purely for decoration, use `alt=""`. See [more tips for alt text](https://a11yproject.com/posts/alt-text/).
- Form fields should always have a corresponding label.
-
- To make sure everything works well with a screenreader, best test it yourself: [NVDA for Windows](https://a11yproject.com/posts/getting-started-with-nvda/), [VoiceOver for macOS](https://a11yproject.com/posts/getting-started-with-voiceover/) or [Universal Access in GNOME](https://help.gnome.org/users/gnome-help/stable/a11y.html)
- If you regularly use a screen reader and there are any problems, don’t hesitate to [open an issue](https://github.com/nextcloud/server/issues/new) :)

## Keyboard navigation

- It should be easy to visually determine the element with keyboard focus. To test, simply use <kbd>Tab ↹</kbd> and make sure all elements are highlighted. Whenever you use `:hover` in your CSS, also include `:focus` and `:active`.
- All elements that can be interacted with via a mouse click should also be possible to interact with via pressing Enter
- Also see: [Quick Test: Navigate using just your keyboard](https://a11yproject.com/posts/navigate-using-just-your-keyboard/)

## Contrast and legibility

- Make sure the text color to background passes WCAG AA recommendation (ideally AAA). You can check it using the [WebAIM color contrast checker](https://webaim.org/resources/contrastchecker/)
- Also see: [Quick Test: Check contrast with your mobile device](https://a11yproject.com/posts/check-contrast-with-mobile-device/)

## Comprehension

- Write in simple and understandable language.
- Don’t use technical jargon which needs prior knowledge. (Like `404 Error`, or `Couldn’t set variable X`) You can do that in the console to help debugging, but don’t show it in the interface.
- Check your written text using [comprehension check tools](https://a11yproject.com/posts/check-comprehension-level/).

## Testing tools

We use various tools for testing like:

- [WAVE browser extensions](https://wave.webaim.org/extension/)
- [Firefox accessibility inspector](https://developer.mozilla.org/en-US/docs/Tools/Accessibility_inspector)
- [Lighthouse for audits of performance, accessibility, and more](https://developers.google.com/web/tools/lighthouse/)
- [Web Accessibility Checklist](https://a11yproject.com/checklist)

## Recommended reading
- [The Accessibility Project](https://a11yproject.com/)
- [Mozilla Developer Network docs on accessibility](https://developer.mozilla.org/en-US/docs/Web/Accessibility)
- [WebAIM articles](https://webaim.org/articles/) and [resources](https://webaim.org/resources/)

@nextcloud/accessibility @nextcloud/designers @MarcoZehe @tyrylu @skjnldsv @kevgathuku @juliushaertl what are additional points we should list? :) I’m wondering how much general accessibility info we should list or rather link to.

**Comment by tyrylu** [comment]
Good question. I have noticed, at least in my daily web usage, that correctly used ARIA roles can also help a lot and can make some on first look too complex interactions perfectly usable.
I think that the existence of ARIA should at least be mentioned.

**Comment by jancborchardt** [comment]
@tyrylu yes of course! :) I wanted to add ARIA too, but a quick glance at the MDN docs only had a site with the role names without explanation. Do you know a good site to link?

**Comment by tyrylu** [comment]
Unfortunately, i looked, but did not find anything terribly useful as well.

**Comment by juliushaertl** [comment]
I would probably link to https://www.w3.org/WAI/tutorials/ which gives a nice overview about the techniques related to the use in web applications. For ARIA maybe just use https://www.w3.org/TR/html-aria/

**Comment by kevgathuku** [comment]
For ARIA, I found this to be a good introduction:
https://developers.google.com/web/fundamentals/accessibility/semantics-aria/

Also the [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.1/) seems to be a good starting point
It's full of examples and here's a snippet from the page:
> It provides guidance on the appropriate application of WAI-ARIA, describes recommended WAI-ARIA usage patterns, and explains concepts behind them.

**Comment by skjnldsv** [comment]
@mariusbluem can you help on this documentation? :)
You're the big doc master! ;)

**Comment by MariusBluem** [comment]
@skjnldsv Let me read some background info ... I‘ll start soon :)

**Comment by jancborchardt** [comment]
The WAI-ARIA page on MDN is also pretty nice: https://developer.mozilla.org/en-US/docs/Learn/Accessibility/WAI-ARIA_basics
